arch-design/multicloud/Multi_Cloud_HA.md

```diff
@@ -0,0 +1,104 @@
+| Scalability            | **Kubernetes acts like more of an all-in-one framework when working with distributed systems. It is thus significantly more complicated as it provides strong guarantees in terms of the cluster state and a unified set of APIs. Container scaling and deployment are therefore slowed down.** | **Docker Swarm can deploy containers much faster than Kubernetes, which allows faster reaction times for scaling on demand**.                                                                                                                                                                   |
+| Container Setup        | By utilizing its own YAML, API, and client definitions, Kubernetes differs from other standard docker equivalents. Thus, Docker Compose or Docker CLI cannot be used to define containers. Also, YAML definitions and commands must be rewritten when switching platforms.                  | The Docker Swarm API offers much of the familiar functionality from Docker, supporting most of the tools that run with Docker. However, Swarm cannot be utilized if the Docker API is deficient in a particular operation.                                                                  |
+| Networking             | Kubernetes has a flat network model, allowing all the pods to communicate with each other. Network policies are in place to define how the pods interact with one another. The network is implemented typically as an overlay, requiring two CIDRS for the services and the pods.           | When a node joins a swarm cluster, it creates an overlay network for services for each host in the docker swarm. It also creates a host-only docker bridge network for containers. This gives users a choice while encrypting the container data traffic to create its own overlay network. |
+| Availability           | **Kubernetes offers significantly high availability as it distributes all the pods among the nodes. This is achieved by absorbing the failure of an application. Unhealthy pods are detected by load balancing services, which subsequently deactivate them**.                                  | Docker also offers high availability architecture since all the services can be cloned in Swarm nodes. The Swarm manager Nodes manage the worker’s node resources and the whole cluster.                                                                                                    |
```

Comment:
How does this impact the client is accessing APIs? Now the APIs are accessed by providing MC node IP in the url. How does the node failure helps the client in availability ?

Reply:
The Clients will be accessing the Multi Cloud services from a Service layer [via floating IP], rather than the fixed IP in previous architecture. Please refer to https://kubernetes.io/docs/concepts/services-networking/service/ for more details.